server/modules/protocol/MySQL/mariadbclient/mysql_client.cc

```diff
@@ -411,6 +411,11 @@ int MySQLSendHandshake(DCB* dcb)
  */
 int gw_MySQLWrite_client(DCB *dcb, GWBUF *queue)
 {
+    MySQLProtocol *protocol = DCB_PROTOCOL(dcb, MySQLProtocol);
+    if (GWBUF_IS_REPLY_OK(queue) && protocol->session_track_trx_state) 
```

Comment:
The DCB has a `service` pointer which can be used to access the service parameters. Storing the `session_track_trx_state` in the service would make this easier.